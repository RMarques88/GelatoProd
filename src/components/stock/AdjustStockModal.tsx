import {
  ActivityIndicator,
  Modal,
  Pressable,
  StyleSheet,
  Text,
  TextInput,
  View,
} from 'react-native';

import { StockMovementType } from '@/domain';

export const movementTypeLabels: Record<StockMovementType, string> = {
  increment: 'Entrada',
  decrement: 'Saída',
  adjustment: 'Ajuste manual',
  initial: 'Estoque inicial',
};

export type AdjustStockModalState = {
  visible: boolean;
  type: StockMovementType;
  quantity: string;
  note: string;
  totalCost: string;
};

export type AdjustStockModalProps = {
  state: AdjustStockModalState;
  onChange: (nextState: AdjustStockModalState) => void;
  onClose: () => void;
  onConfirm: () => void;
  isSubmitting: boolean;
  disabled?: boolean;
};

export function AdjustStockModal({
  state,
  onChange,
  onClose,
  onConfirm,
  isSubmitting,
  disabled = false,
}: AdjustStockModalProps) {
  const shouldCaptureCost = state.type === 'increment' || state.type === 'initial';

  return (
    <Modal
      visible={state.visible}
      animationType="slide"
      transparent
      onRequestClose={onClose}
    >
      <View style={styles.modalOverlay}>
        <View style={styles.modalContent}>
          <View style={styles.modalHeader}>
            <Text style={styles.modalTitle}>Registrar movimentação</Text>
            <Pressable
              onPress={onClose}
              style={({ pressed }) => [
                styles.modalCloseButton,
                pressed && styles.modalCloseButtonPressed,
              ]}
            >
              <Text style={styles.modalCloseText}>Fechar</Text>
            </Pressable>
          </View>

          <View style={styles.modalBody}>
            <Text style={styles.modalLabel}>Tipo de movimentação</Text>
            <View style={styles.modalToggleGroup}>
              {(['increment', 'decrement', 'adjustment'] as StockMovementType[]).map(
                option => (
                  <Pressable
                    key={option}
                    onPress={() => onChange({ ...state, type: option })}
                    style={({ pressed }) => [
                      styles.modalToggleButton,
                      state.type === option && styles.modalToggleButtonActive,
                      pressed && styles.modalToggleButtonPressed,
                    ]}
                    disabled={disabled}
                  >
                    <Text
                      style={[
                        styles.modalToggleButtonText,
                        state.type === option && styles.modalToggleButtonTextActive,
                      ]}
                    >
                      {movementTypeLabels[option]}
                    </Text>
                  </Pressable>
                ),
              )}
            </View>

            <Text style={styles.modalLabel}>Quantidade (em gramas)</Text>
            <TextInput
              value={state.quantity}
              onChangeText={value => onChange({ ...state, quantity: value })}
              placeholder="500"
              style={styles.modalInput}
              keyboardType="numeric"
              editable={!disabled}
            />
            <Text style={styles.modalHintText}>
              As movimentações são registradas em gramas, independente da unidade padrão
              do produto.
            </Text>

            {shouldCaptureCost ? (
              <View style={styles.modalFieldGroup}>
                <Text style={styles.modalLabel}>Valor total da compra (R$)</Text>
                <TextInput
                  value={state.totalCost}
                  onChangeText={value => onChange({ ...state, totalCost: value })}
                  placeholder="150,00"
                  style={styles.modalInput}
                  keyboardType="numeric"
                  editable={!disabled}
                />
              </View>
            ) : null}

            <Text style={styles.modalLabel}>Observações</Text>
            <TextInput
              value={state.note}
              onChangeText={value => onChange({ ...state, note: value })}
              placeholder="Motivo do ajuste (opcional)"
              style={[styles.modalInput, styles.modalInputMultiline]}
              multiline
              numberOfLines={3}
              editable={!disabled}
            />
          </View>

          <Pressable
            onPress={onConfirm}
            style={({ pressed }) => [
              styles.modalPrimaryButton,
              pressed && styles.modalPrimaryButtonPressed,
            ]}
            disabled={disabled || isSubmitting}
          >
            {isSubmitting ? (
              <ActivityIndicator color="#FFFFFF" />
            ) : (
              <Text style={styles.modalPrimaryButtonText}>Confirmar ajuste</Text>
            )}
          </Pressable>
        </View>
      </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  modalOverlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.45)',
    justifyContent: 'flex-end',
  },
  modalContent: {
    backgroundColor: '#FFFFFF',
    borderTopLeftRadius: 24,
    borderTopRightRadius: 24,
    padding: 24,
    gap: 16,
  },
  modalHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  modalTitle: {
    fontSize: 18,
    fontWeight: '700',
    color: '#1F2937',
  },
  modalCloseButton: {
    paddingHorizontal: 12,
    paddingVertical: 6,
    borderRadius: 8,
  },
  modalCloseButtonPressed: {
    backgroundColor: '#F3F4F6',
  },
  modalCloseText: {
    fontSize: 14,
    fontWeight: '600',
    color: '#4B5563',
  },
  modalBody: {
    gap: 12,
  },
  modalLabel: {
    fontSize: 14,
    fontWeight: '600',
    color: '#1F2937',
  },
  modalFieldGroup: {
    gap: 8,
  },
  modalToggleGroup: {
    flexDirection: 'row',
    gap: 12,
  },
  modalToggleButton: {
    flex: 1,
    paddingVertical: 12,
    borderRadius: 12,
    borderWidth: 1,
    borderColor: '#D1D5DB',
    backgroundColor: '#F3F4F6',
    alignItems: 'center',
  },
  modalToggleButtonActive: {
    backgroundColor: '#D1FAE5',
    borderColor: '#10B981',
  },
  modalToggleButtonPressed: {
    opacity: 0.85,
  },
  modalToggleButtonText: {
    fontSize: 14,
    fontWeight: '600',
    color: '#1F2937',
  },
  modalToggleButtonTextActive: {
    color: '#047857',
  },
  modalInput: {
    borderWidth: 1,
    borderColor: '#D1D5DB',
    borderRadius: 12,
    paddingHorizontal: 14,
    paddingVertical: 12,
    fontSize: 15,
    color: '#111827',
    backgroundColor: '#FFFFFF',
  },
  modalInputMultiline: {
    minHeight: 96,
    textAlignVertical: 'top',
  },
  modalHintText: {
    fontSize: 12,
    color: '#6B7280',
  },
  modalPrimaryButton: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 14,
    borderRadius: 12,
    backgroundColor: '#4E9F3D',
  },
  modalPrimaryButtonPressed: {
    opacity: 0.9,
  },
  modalPrimaryButtonText: {
    color: '#FFFFFF',
    fontSize: 16,
    fontWeight: '700',
  },
});
