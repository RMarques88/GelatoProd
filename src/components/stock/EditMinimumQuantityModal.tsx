import {
  ActivityIndicator,
  KeyboardAvoidingView,
  Modal,
  Platform,
  Pressable,
  StyleSheet,
  Text,
  TextInput,
  View,
} from 'react-native';

export type EditMinimumQuantityModalProps = {
  visible: boolean;
  value: string;
  productName?: string | null;
  errorMessage?: string | null;
  onChangeValue: (value: string) => void;
  onClose: () => void;
  onConfirm: () => void;
  isSubmitting: boolean;
  disabled?: boolean;
};

export function EditMinimumQuantityModal({
  visible,
  value,
  productName,
  errorMessage,
  onChangeValue,
  onClose,
  onConfirm,
  isSubmitting,
  disabled = false,
}: EditMinimumQuantityModalProps) {
  return (
    <Modal visible={visible} transparent animationType="slide" onRequestClose={onClose}>
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : undefined}
        style={styles.overlay}
      >
        <Pressable style={styles.backdrop} onPress={onClose} />
        <View style={styles.content}>
          <View style={styles.header}>
            <View>
              <Text style={styles.title}>Editar quantidade mínima</Text>
              {productName ? <Text style={styles.subtitle}>{productName}</Text> : null}
            </View>
            <Pressable
              onPress={onClose}
              style={({ pressed }) => [
                styles.closeButton,
                pressed && styles.closeButtonPressed,
              ]}
            >
              <Text style={styles.closeButtonText}>Fechar</Text>
            </Pressable>
          </View>

          <View style={styles.body}>
            <Text style={styles.label}>Quantidade mínima (g)</Text>
            <TextInput
              value={value}
              onChangeText={onChangeValue}
              keyboardType="numeric"
              placeholder="Ex: 1500"
              style={styles.input}
              editable={!disabled}
            />
            <Text style={styles.hint}>
              Essa quantidade será usada para gerar alertas quando o estoque estiver
              baixo.
            </Text>
            {errorMessage ? <Text style={styles.error}>{errorMessage}</Text> : null}
          </View>

          <View style={styles.actions}>
            <Pressable
              onPress={onClose}
              style={({ pressed }) => [
                styles.secondaryButton,
                pressed && styles.secondaryButtonPressed,
              ]}
              disabled={isSubmitting}
            >
              <Text style={styles.secondaryButtonText}>Cancelar</Text>
            </Pressable>
            <Pressable
              onPress={onConfirm}
              style={({ pressed }) => [
                styles.primaryButton,
                pressed && styles.primaryButtonPressed,
              ]}
              disabled={disabled || isSubmitting}
            >
              {isSubmitting ? (
                <ActivityIndicator color="#FFFFFF" />
              ) : (
                <Text style={styles.primaryButtonText}>Atualizar mínimo</Text>
              )}
            </Pressable>
          </View>
        </View>
      </KeyboardAvoidingView>
    </Modal>
  );
}

const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    justifyContent: 'flex-end',
    backgroundColor: 'rgba(0, 0, 0, 0.45)',
  },
  backdrop: {
    ...StyleSheet.absoluteFillObject,
  },
  content: {
    backgroundColor: '#FFFFFF',
    borderTopLeftRadius: 24,
    borderTopRightRadius: 24,
    paddingHorizontal: 24,
    paddingTop: 24,
    paddingBottom: 32,
    gap: 20,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  title: {
    fontSize: 18,
    fontWeight: '700',
    color: '#1F2937',
  },
  subtitle: {
    marginTop: 4,
    fontSize: 14,
    color: '#4B5563',
  },
  closeButton: {
    paddingHorizontal: 12,
    paddingVertical: 6,
    borderRadius: 10,
  },
  closeButtonPressed: {
    backgroundColor: '#F3F4F6',
  },
  closeButtonText: {
    fontSize: 14,
    fontWeight: '600',
    color: '#4B5563',
  },
  body: {
    gap: 12,
  },
  label: {
    fontSize: 14,
    fontWeight: '600',
    color: '#1F2937',
  },
  input: {
    borderWidth: 1,
    borderColor: '#D1D5DB',
    borderRadius: 12,
    paddingHorizontal: 16,
    paddingVertical: 12,
    fontSize: 16,
    color: '#111827',
    backgroundColor: '#FFFFFF',
  },
  hint: {
    fontSize: 12,
    color: '#6B7280',
  },
  error: {
    fontSize: 13,
    color: '#DC2626',
  },
  actions: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    gap: 12,
  },
  secondaryButton: {
    paddingHorizontal: 18,
    paddingVertical: 12,
    borderRadius: 12,
    borderWidth: 1,
    borderColor: '#D1D5DB',
    backgroundColor: '#FFFFFF',
  },
  secondaryButtonPressed: {
    backgroundColor: '#F3F4F6',
  },
  secondaryButtonText: {
    fontSize: 14,
    fontWeight: '600',
    color: '#1F2937',
  },
  primaryButton: {
    paddingHorizontal: 18,
    paddingVertical: 12,
    borderRadius: 12,
    backgroundColor: '#2563EB',
    minWidth: 160,
    alignItems: 'center',
  },
  primaryButtonPressed: {
    opacity: 0.85,
  },
  primaryButtonText: {
    fontSize: 14,
    fontWeight: '700',
    color: '#FFFFFF',
  },
});
