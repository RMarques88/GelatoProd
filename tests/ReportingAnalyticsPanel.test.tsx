import React from 'react';
import renderer, { act } from 'react-test-renderer';

import { ReportingAnalyticsPanel } from '@/components/stock/ReportingAnalyticsPanel';
import type {
  DivergenceUsagePeriodSummary,
  IngredientConsumptionPeriodSummary,
  RecipeProductionPeriodSummary,
} from '@/services/reportingMetrics';

describe('ReportingAnalyticsPanel', () => {
  const recipeSummaries: RecipeProductionPeriodSummary[] = [
    {
      periodKey: 'production_semana_40',
      periodLabel: 'Semana 40 (01 out – 07 out)',
      periodStart: new Date('2025-10-01T00:00:00.000Z'),
      periodEnd: new Date('2025-10-07T23:59:59.000Z'),
      totals: {
        totalPlans: 3,
        totalPlannedQuantityInUnits: 260,
        totalActualQuantityInUnits: 248,
        totalEstimatedCostInBRL: 1310,
        totalActualCostInBRL: 1280,
        recipes: [
          {
            recipeId: 'recipe-vanilla',
            recipeName: 'Sorvete de Baunilha',
            planCount: 2,
            plannedQuantityInUnits: 180,
            actualQuantityInUnits: 170,
            estimatedCostInBRL: 890,
            actualCostInBRL: 870,
          },
          {
            recipeId: 'recipe-pistachio',
            recipeName: 'Gelato de Pistache',
            planCount: 1,
            plannedQuantityInUnits: 80,
            actualQuantityInUnits: 78,
            estimatedCostInBRL: 420,
            actualCostInBRL: 410,
          },
        ],
      },
    },
  ];

  const ingredientSummaries: IngredientConsumptionPeriodSummary[] = [
    {
      periodKey: 'consumption_semana_40',
      periodLabel: 'Semana 40 (01 out – 07 out)',
      periodStart: new Date('2025-10-01T00:00:00.000Z'),
      periodEnd: new Date('2025-10-07T23:59:59.000Z'),
      totals: {
        totalMovements: 4,
        totalConsumedInGrams: 3200,
        products: [
          {
            productId: 'prod-milk',
            quantityInGrams: 1800,
            movementCount: 2,
          },
          {
            productId: 'prod-sugar',
            quantityInGrams: 900,
            movementCount: 1,
          },
          {
            productId: 'prod-cream',
            quantityInGrams: 500,
            movementCount: 1,
          },
        ],
      },
    },
  ];

  const divergenceSummaries: DivergenceUsagePeriodSummary[] = [
    {
      periodKey: 'divergence_semana_40',
      periodLabel: 'Semana 40 (01 out – 07 out)',
      periodStart: new Date('2025-10-01T00:00:00.000Z'),
      periodEnd: new Date('2025-10-07T23:59:59.000Z'),
      totals: {
        totalDivergences: 2,
        totalShortageInUnits: 35,
        severity: [
          {
            severity: 'high',
            count: 1,
            totalShortageInUnits: 25,
          },
          {
            severity: 'medium',
            count: 1,
            totalShortageInUnits: 10,
          },
        ],
      },
    },
  ];

  const productNames = new Map([
    ['prod-milk', 'Leite integral'],
    ['prod-sugar', 'Açúcar refinado'],
    ['prod-cream', 'Creme de leite'],
  ]);

  it('renders analytics data snapshot', () => {
    let testRenderer: renderer.ReactTestRenderer;

    act(() => {
      testRenderer = renderer.create(
        React.createElement(ReportingAnalyticsPanel, {
          granularity: 'week',
          rangeLabel: '01 out – 07 out',
          isLoading: false,
          error: null,
          recipeSummaries,
          ingredientSummaries,
          divergenceSummaries,
          onRetry: jest.fn(),
          getProductName: productId =>
            productNames.get(productId) ?? 'Produto desconhecido',
        }),
      );
    });

    expect(testRenderer!.toJSON()).toMatchSnapshot();
    act(() => {
      testRenderer!.unmount();
    });
  });

  it('renders error state snapshot', () => {
    let testRenderer: renderer.ReactTestRenderer;

    act(() => {
      testRenderer = renderer.create(
        React.createElement(ReportingAnalyticsPanel, {
          granularity: 'week',
          rangeLabel: '01 out – 07 out',
          isLoading: false,
          error: new Error('Firestore indisponível'),
          recipeSummaries: [],
          ingredientSummaries: [],
          divergenceSummaries: [],
          onRetry: jest.fn(),
          getProductName: () => 'Produto desconhecido',
        }),
      );
    });

    expect(testRenderer!.toJSON()).toMatchSnapshot();
    act(() => {
      testRenderer!.unmount();
    });
  });
});
