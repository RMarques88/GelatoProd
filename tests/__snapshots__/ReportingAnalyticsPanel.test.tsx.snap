// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportingAnalyticsPanel renders analytics data snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 16,
      "elevation": 3,
      "marginBottom": 32,
      "marginTop": 24,
      "padding": 20,
      "shadowColor": "#111827",
      "shadowOffset": {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.06,
      "shadowRadius": 12,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 16,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "color": "#111827",
            "fontSize": 18,
            "fontWeight": "700",
          }
        }
      >
        Tendências analíticas
      </Text>
      <Text
        style={
          {
            "color": "#6B7280",
            "fontSize": 13,
            "marginTop": 4,
          }
        }
      >
        Semana
         · 
        01 out – 07 out
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "column",
        "gap": 16,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#F9FAFB",
          "borderRadius": 14,
          "gap": 12,
          "padding": 16,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
            {
              "backgroundColor": "#E0F2FE",
            },
          ]
        }
      >
        <Icon
          name="bar-chart-outline"
        />
      </View>
      <Text
        style={
          {
            "color": "#1F2937",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Produção por receita
      </Text>
      <Text
        style={
          {
            "color": "#6B7280",
            "fontSize": 13,
          }
        }
      >
        Semana 40 (01 out – 07 out)
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 12,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "flex": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            3
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
                "marginTop": 4,
              }
            }
          >
            Planos concluídos
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "flex": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            260
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
                "marginTop": 4,
              }
            }
          >
            Qtd planejada
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "flex": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            248
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
                "marginTop": 4,
              }
            }
          >
            Qtd real
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 12,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "flex": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            R$ 1.310,00
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
                "marginTop": 4,
              }
            }
          >
            Custo estimado
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "flex": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            R$ 1.280,00
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
                "marginTop": 4,
              }
            }
          >
            Custo real
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#E5E7EB",
            "height": 0.5,
          }
        }
      />
      <View
        style={
          {
            "gap": 10,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "gap": 2,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            Sorvete de Baunilha
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
              }
            }
          >
            2
             plano(s) ·
             
            170
             
            un reais
          </Text>
          <Text
            style={
              {
                "color": "#374151",
                "fontSize": 12,
                "marginTop": 2,
              }
            }
          >
            Custo: 
            R$ 870,00
             
            (R$ 890,00 previsto)
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "gap": 2,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            Gelato de Pistache
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
              }
            }
          >
            1
             plano(s) ·
             
            78
             
            un reais
          </Text>
          <Text
            style={
              {
                "color": "#374151",
                "fontSize": 12,
                "marginTop": 2,
              }
            }
          >
            Custo: 
            R$ 410,00
             
            (R$ 420,00 previsto)
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#F9FAFB",
          "borderRadius": 14,
          "gap": 12,
          "padding": 16,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
            {
              "backgroundColor": "#FEF3C7",
            },
          ]
        }
      >
        <Icon
          name="leaf-outline"
        />
      </View>
      <Text
        style={
          {
            "color": "#1F2937",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Consumo de insumos
      </Text>
      <Text
        style={
          {
            "color": "#6B7280",
            "fontSize": 13,
          }
        }
      >
        Semana 40 (01 out – 07 out)
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 12,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "flex": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            4
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
                "marginTop": 4,
              }
            }
          >
            Movimentações
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "flex": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            3.200 g
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
                "marginTop": 4,
              }
            }
          >
            Consumo total
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#E5E7EB",
            "height": 0.5,
          }
        }
      />
      <View
        style={
          {
            "gap": 10,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "gap": 2,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            Leite integral
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
              }
            }
          >
            1.800 g · 2 mov.
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "gap": 2,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            Açúcar refinado
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
              }
            }
          >
            900 g · 1 mov.
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "gap": 2,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            Creme de leite
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
              }
            }
          >
            500 g · 1 mov.
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#F9FAFB",
          "borderRadius": 14,
          "gap": 12,
          "padding": 16,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
            {
              "backgroundColor": "#EDE9FE",
            },
          ]
        }
      >
        <Icon
          name="alert-circle-outline"
        />
      </View>
      <Text
        style={
          {
            "color": "#1F2937",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Divergências de produção
      </Text>
      <Text
        style={
          {
            "color": "#6B7280",
            "fontSize": 13,
          }
        }
      >
        Semana 40 (01 out – 07 out)
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 12,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "flex": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            2
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
                "marginTop": 4,
              }
            }
          >
            Registros
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "flex": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#111827",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            35
          </Text>
          <Text
            style={
              {
                "color": "#6B7280",
                "fontSize": 12,
                "marginTop": 4,
              }
            }
          >
            Faltas acumuladas (un)
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#E5E7EB",
            "height": 0.5,
          }
        }
      />
      <View
        style={
          {
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 8,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 4,
                  "height": 8,
                  "width": 8,
                },
                {
                  "backgroundColor": "#DC2626",
                },
              ]
            }
          />
          <Text
            style={
              {
                "color": "#4B5563",
                "fontSize": 12,
              }
            }
          >
            HIGH
            : 
            1
             registro(s)
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 8,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 4,
                  "height": 8,
                  "width": 8,
                },
                {
                  "backgroundColor": "#F59E0B",
                },
              ]
            }
          />
          <Text
            style={
              {
                "color": "#4B5563",
                "fontSize": 12,
              }
            }
          >
            MEDIUM
            : 
            1
             registro(s)
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`ReportingAnalyticsPanel renders error state snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 16,
      "elevation": 3,
      "marginBottom": 32,
      "marginTop": 24,
      "padding": 20,
      "shadowColor": "#111827",
      "shadowOffset": {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.06,
      "shadowRadius": 12,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 16,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "color": "#111827",
            "fontSize": 18,
            "fontWeight": "700",
          }
        }
      >
        Tendências analíticas
      </Text>
      <Text
        style={
          {
            "color": "#6B7280",
            "fontSize": 13,
            "marginTop": 4,
          }
        }
      >
        Semana
         · 
        01 out – 07 out
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FEF2F2",
        "borderColor": "#FCA5A5",
        "borderRadius": 12,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "gap": 12,
        "marginBottom": 16,
        "padding": 16,
      }
    }
  >
    <Icon
      name="alert-circle-outline"
    />
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#991B1B",
            "fontSize": 14,
            "fontWeight": "600",
          }
        }
      >
        Não foi possível carregar as métricas.
      </Text>
      <Text
        style={
          {
            "color": "#B91C1C",
            "fontSize": 13,
            "marginTop": 2,
          }
        }
      >
        Firestore indisponível
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "#111827",
            "borderRadius": 999,
            "paddingHorizontal": 14,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 13,
            "fontWeight": "600",
          }
        }
      >
        Tentar novamente
      </Text>
    </View>
  </View>
</View>
`;
